src/components/App.tsx

```diff
@@ -2055,7 +2055,9 @@ export class App extends React.Component<any, AppState> {
                 mutateElement(element, {
                   width,
                   height,
-                  ...adjustXYWithRotation("nw", element, deltaX, dY, angle),
+                  ...(isLinearElement(element) && (width < 0 || height < 0)
+                    ? element
```

Comment:
Good catch!

```suggestion
                    ? { x: element.x, y: element.y }
```
It's better to pass only `x` and `y`. Sorry, it was not very intuitive.

Looking at the code again, would it better to avoid calling `mutateElement` at all if `width` or `height` is negative?

Reply:
@dai-shi Thanks for the review!

> It's better to pass only x and y.

I fixed it.

> would it better to avoid calling mutateElement at all if width or height is negative?

I'm not sure because I'm still not familiar with this app...

Reply:
> I'm not sure because I'm still not familiar with this app...

I think it will work. While we wait others to review this PR, would you mind trying?

We could refactor further something like:
```ts
            case "nw":
              const width = element.width - deltaX;
              const height = event.shiftKey ? width : element.height - deltaY;
              const dY = element.height - height;
              if (isLinearElement(element)) {
                if (element.points.length === 2) {
                  const [, p1] = element.points;
                  if (!resizeArrowFn) {
                    if (p1[0] < 0 || p1[1] < 0) {
                      resizeArrowFn = arrowResizeEnd;
                    } else {
                      resizeArrowFn = arrowResizeOrigin;
                    }
                  }
                  resizeArrowFn(element, 1, deltaX, deltaY, x, y, event.shiftKey);
                } else if (width >= 0 && height >=0) {
                  mutateElement(element, {
                    width,
                    height,
                    ...(adjustXYWithRotation("nw", element, deltaX, dY, angle)),
                    points: rescalePoints(
                      0,
                      width,
                      rescalePoints(1, height, element.points),
                    ),
                  });
                }
              } else {
                mutateElement(element, {
                  width,
                  height,
                  ...(adjustXYWithRotation("nw", element, deltaX, dY, angle)),
                });
              }
              break;
```

We could even start with `if (!isLinearElement(element)) {`, which may simplify more.

Reply:
Thank you, I understand. It looks good.